### DISCUSSION TITLE
How do you run the unittests locally?

### DISCUSSION THREAD

**Discussion by ColinKennedy** [nvim-treesitter/nvim-treesitter#4768] [Q&A]
I'd like to add a new parser but before I send out a PR, I would like to make sure the highlights.scm unittests will pass so I don't waste any PR reviewer's time. By checking out the https://github.com/nvim-treesitter/nvim-treesitter/blob/master/.github/workflows/tests.yml file, I see `PATH=/usr/local/bin:$PATH ./scripts/run_tests.sh` but it errors with `Not an editor command: PlenaryBustedDirectory`. If I try to run the command that it's running manually, 

```
    nvim --headless --noplugin -u scripts/minimal_init.lua \
        -c "PlenaryBustedDirectory $1 { minimal_init = './scripts/minimal_init.lua' }"
```

And replace `$1` with `tests/query` then I get a stdout of

```
Starting...
Scheduling: tests/query/highlights_spec.lua
Scheduling: tests/query/injection_spec.lua
```

And eventually it ends without printing anything.

I'm guessing the lack of output is because my host machine isn't set up with the same assumptions as the GitHub action's container is (I use CentOS 7, the action is using ubuntu).

What I'm trying to do is test my specific contribution's highlights and indent unittests. Ideally I'd like to simply do `test ./tests/indent/usd` / `highlight-assertions ./tests/query/highlights/usd` from the terminal and I'm guessing that you can. I just couldn't find it in the documentation yet. Could someone please lend a hand and point me in the right direction? Thank you!

**Comment by NoahTheDuke** [comment]
I opened [this issue](https://github.com/nvim-treesitter/nvim-treesitter/issues/3397) last year. Didn't get much traction but might lead somewhere useful.
